FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Multitask/DemoTasks/mqtt-agent-task.c

```diff
@@ -0,0 +1,955 @@
+/*
+ * FreeRTOS V202012.00
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * https://www.FreeRTOS.org
+ * https://github.com/FreeRTOS
+ *
+ */
+
+/*
+ * This demo creates multiple tasks, all of which use the MQTT agent API to
+ * communicate with an MQTT broker through the same MQTT connection.
+ *
+ * This file contains the initial task created after the TCP/IP stack connects
+ * to the network.  The task:
+ *
+ * 1) Connects to the MQTT broker.
+ * 2) Creates the other demo tasks, in accordance with the #defines set in
+ *    demo_config.h.  For example, if demo_config.h contains the following
+ *    settings:
+ *
+ *    #define democonfigCREATE_LARGE_MESSAGE_SUB_PUB_TASK     1
+ *    #define democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE 3
+ *
+ *    then the initial task will create the task implemented in
+ *    LargeMessageSubScribePublish.c and three instances of the task
+ *    implemented in SimpleSubscribePublishT.c.  See the comments at the top
+ *    of those files for more information.
```

Comment:
The second point needs to be updated to not mention about the `democonfigCREATE_LARGE_MESSAGE_SUB_PUB_TASK`
 config but more specifically mention about the number of Pub Sub tasks it will create depending on the `democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE` config

Reply:
Updated